scipy/stats/tests/test_resampling.py

```diff
@@ -732,124 +737,188 @@ def statistic_extradim(*args, axis):
 class TestMonteCarloHypothesisTest:
     atol = 2.5e-2  # for comparing p-value
 
-    def rvs(self, rvs_in, rs):
-        return lambda *args, **kwds: rvs_in(*args, random_state=rs, **kwds)
+    def get_rvs(self, rvs_in, rs, xp=np):
+        return lambda *args, **kwds: xp.asarray(rvs_in(*args, random_state=rs, **kwds))
 
-    def test_input_validation(self):
+    def get_statistic(self, xp):
+        def statistic(x, axis):
+            m = xp.mean(x, axis=axis)
+            v = xp.var(x, axis=axis, correction=1)
+            n = x.shape[axis]
+            return m / (v/n)**0.5
+            # return stats.ttest_1samp(x, popmean=0., axis=axis).statistic)
```

Comment:
This works for CPU, but I wanted to test GPU, too.

Reply:
Has this been done?

Reply:
Yup, I was testing with CuPy when I wrote this.